inc/back-compat.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Twenty Twenty-Two: Backwards compatibility functionality
+ *
+ * Prevents the theme from running on WordPress versions prior to 5.8,
+ * since this theme is not meant to be backward compatible beyond that and
+ * relies on many newer functions and markup changes introduced in 5.8.
+ *
+ * @since Twenty Twenty-Two 1.0
+ */
+
+/**
+ * Display upgrade notice on theme switch.
+ *
+ * @since Twenty Twenty-Two 1.0
+ *
+ * @return void
+ */
+function twentytwentytwo_switch_theme() {
+	add_action( 'admin_notices', 'twentytwentytwo_upgrade_notice' );
+}
+add_action( 'after_switch_theme', 'twentytwentytwo_switch_theme' );
+
+/**
+ * Adds a message for unsuccessful theme switch.
+ *
+ * Prints an update nag after an unsuccessful attempt to switch to
+ * the theme on WordPress versions prior to 5.8.
+ *
+ * @since Twenty Twenty-Two 1.0
+ *
+ * @global string $wp_version WordPress version.
+ *
+ * @return void
+ */
+function twentytwentytwo_upgrade_notice() {
+	echo '<div class="error"><p>';
+	printf(
+	/* translators: %s: WordPress Version. */
+		esc_html__( 'This theme requires WordPress 5.8 or newer. You are running version %s. Please upgrade.', 'twentytwentytwo' ),
+		esc_html( $GLOBALS['wp_version'] )
+	);
+	echo '</p></div>';
+}
+
+/**
+ * Prevents the Customizer from being loaded on WordPress versions prior to 5.8.
```

Comment:
@jffng and I plan on changing the required version to 5.9 for release. We've been keeping it at 5.8 for now while folks are still testing on that version. (In general, it won't run on 5.8 without the Gutenberg plugin, so we feel that 5.9 is a more honest minimum version).

Reply:
Yes, I had assumed it would only work on 5.9+ but then noticed that the README said 5.8+, so I sticked with that for now. Using 5.9 now would probably make it untestable on current trunk.